app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.java

```diff
@@ -48,4 +52,36 @@ protected void updateDefaultItem(CompoundButton defaultButton) {
         }
         return engineIdSet;
     }
+
+    // Whenever an engine is checked or unchecked, we notify the menu
+    public void bindEngineCheckboxesToMenu() {
+        for (int i = 0; i < searchEngineGroup.getChildCount(); i++) {
+            final CompoundButton engineButton = (CompoundButton) searchEngineGroup.getChildAt(i);
+            engineButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                    ((Activity) getContext()).invalidateOptionsMenu();
```

Comment:
Nice removing the binding to the menu item - however, I'm now seeing a flicker on every checkbox click because we're invalidating every time the items are clicked.

We should only invalidate in two cases:
1) if the button is newly checked, and it is the ONLY item checked
2) if the button is newly unchecked, and there are NO items checked

so something like:
```
if (isChecked && numItemsChecked() == 1) || (!isChecked && numItemsChecked() == 0) {
    invalidateOptionsMenu()
}
```
(sorry for writing the code already! but I didn't want to make you do something impossible x_x and then the code just happened)